dockerclient/client.go

```diff
@@ -923,6 +923,28 @@ func (e *ClientExecutor) Archive(fromFS bool, src, dst string, allowDownload boo
 		klog.V(5).Infof("Archiving %s %s -> %s from context archive", e.ContextArchive, src, dst)
 		return archiveFromFile(e.ContextArchive, src, dst, excludes, check)
 	}
+	// if the source is an archive, extract it under the destination directory
+	srcFullPath := filepath.Join(e.Directory, src)
+	if isArchivePath(srcFullPath) {
+		f, err := os.Open(srcFullPath)
+		if err != nil {
+			return nil, nil, fmt.Errorf("error opening %q: %v", srcFullPath, err)
+		}
+		klog.V(5).Infof("Extracting %s -> %s from local archive", srcFullPath, dst)
+		transform := func(h *tar.Header, r io.Reader) (data []byte, update bool, skip bool, err error) {
+			h.Name = filepath.Join(dst, h.Name)
+			if h.Typeflag == tar.TypeLink {
+				h.Linkname = filepath.Join(dst, h.Linkname)
+			}
+			return nil, false, false, nil
+		}
+		filtered, err := transformArchive(f, true, transform)
```

Comment:
might be end of day brain, but where does `transformArchive` live?

Reply:
It's in [archive.go](https://github.com/openshift/imagebuilder/blob/master/dockerclient/archive.go#L234).  This part of imagebuilder is the direct inspiration for similar functionality [in buildah](https://github.com/containers/buildah/blob/master/digester.go#L90).